javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/LambdaExprContext.java

```diff
@@ -66,9 +67,10 @@ public LambdaExprContext(LambdaExpr wrappedNode, TypeSolver typeSolver) {
 
     @Override
     public Optional<Value> solveSymbolAsValue(String name) {
+        int index = -1;
         for (Parameter parameter : wrappedNode.getParameters()) {
+            index++;
```

Comment:
I chose to initialize index to -1 and do the update immediately to make it more obvious what the index is referring to. Since this method is quite long, updating index at the end made it difficult to connect the value of index to what was happening in code.

Reply:
Hi @johannescoetzee I don't see what fixes the bug you are trying to solve.

Reply:
The only thing this PR really changes is that it moves the declaration and initialization of `index` out of the `for(Parameter ...)` loop.

Before, `index` was always set to 0 inside the loop, so when it got to the statement
```
ResolvedType lambdaType = functionalMethod.get().getParamType(index);
```
on line 154, `lambdaType` was always set to the type of param 0 (hence the bug). By initializing `index` before the loop, it's now correctly incremented as parameters are checked.